bitcoin/src/electrs.rs

```diff
@@ -0,0 +1,19 @@
+use crate::Error;
+
+async fn get(base_path: &str, url: &str) -> Result<String, Error> {
+    let response = reqwest::get(format!("{}/{}", base_path, url)).await?;
+    let status = response.status();
+    if status.is_client_error() || status.is_server_error() {
+        Err(Error::ElectrsQueryFailed)
+    } else {
+        Ok(response.text().await?)
+    }
+}
+
+pub async fn get_tx_hex(base_path: &str, tx_id: &str) -> Result<String, Error> {
```

Comment:
Unfortunately the code generated by the OpenAPI spec always uses `serde_json::from_str` regardless of the actual return type and I couldn't see a way to change this behaviour. This fails to decode a `String` since it is not valid JSON and there is no way to patch the library right now since the code is generated in the CI.